src/main/groovy/edu/oregonstate/mist/inventory/db/InventoryDAOWrapper.groovy

```diff
@@ -112,15 +115,112 @@ class InventoryDAOWrapper {
     }
 
     /**
+     * Update a single inventory object by ID
+     * @param inventory
+     * @param inventoryID
+     */
+    @Transaction
+    public void updateInventory(Inventory inventory, String inventoryID) {
+        //Update top level inventory attributes
+        inventory.trimNameAndDescription()
+        inventoryDAO.updateInventory(inventory, inventoryID)
+
+        //Update api query parameters
+        createUpdateDeleteFields(inventory.apiQueryParams, inventoryID, inventoryID, QUERY_DB_TYPE)
+
+        //Update consuming entities
+        List<ConsumingEntity> currentConsumingEntities =
+                inventoryDAO.getConsumingEntities(inventoryID)
+
+        def currentConsumingEntityIDs = currentConsumingEntities.collect { it.entityID }
+
+        inventory.consumingEntities.each { consumingEntity ->
```

Comment:
A small suggestion would be to use put some logger.debug messages to help with troubleshooting down the road.